src/metaopt/worker/experiment.py

```diff
@@ -0,0 +1,354 @@
+            is_new = True
+        else:
+            # Fork if it is needed
+            is_new = self._is_different_from(config)
+            if is_new:
+                self._fork_config(config)  # Change (?) `name` attribute here.
+
+        # Just overwrite everything given
+        for section, value in six.iteritems(config):
+            if section == 'status' or \
+                    section not in self.__slots__ or \
+                    section.startswith('_'):
+                continue
+            setattr(self, section, value)
+
+        self.status = 'new'
+        final_config = self.configuration  # grab dict representation of Experiment
+
+        # Sanitize and replace some sections with objects
+        self._sanitize_config()
+
+        # If everything is alright, push new config to database
+        if is_new:
+            self._db.write('experiments', final_config)
+            # XXX: This may be MongoDB only; it updates an inserted dict with _id
+            self._id = final_config['_id']
+        else:
+            self._db.write('experiments', final_config, {'_id': self._id})
+
+        self._init_done = True
+
+    @property
+    def stats(self):
+        """Calculate a stats dictionary for this particular experiment.
+
+        Returns
+        -------
+        stats : dict
+
+        Stats
+        -----
+        trials_completed : int
+           Number of completed trials
+        best_trials_id : int
+           Unique identifier of the `Trial` object in the database which achieved
+           the best known objective result.
+        best_evaluation : float
+           Evaluation score of the best trial
+        start_time : `datetime.datetime`
+           When Experiment was first dispatched and started running.
+        finish_time : `datetime.datetime`
+           When Experiment reached terminating condition and stopped running.
+        duration : `datetime.timedelta`
+           Elapsed time.
+
+        """
+        query = dict(
+            exp_name=self.name,
+            user=self.metadata['user'],
+            status='completed'
+            )
+        completed_trials = self._db.read('trials', query,
+                                         selection={'_id': 1, 'end_time': 1,
+                                                    'results': 1})
+        stats = dict()
+        stats['trials_completed'] = len(completed_trials)
+        stats['best_trials_id'] = None
+        stats['best_evaluation'] = math.inf
+        stats['start_time'] = self.metadata['datetime']
+        stats['finish_time'] = stats['start_time']
+        for trial in completed_trials:
+            if trial.end_time > stats['finish_time']:
+                stats['finish_time'] = trial.end_time
+            assert trial.results[0].type == 'objective'
+            objective = trial.results[0].value
+            if objective < stats['best_evaluation']:
```

Comment:
Indeed let's decide if we are going to support both min and max at the MVP. What do you think?

Reply:
I think that's fundamental enough for the MVP, but we can fix this in another PR later on so it gives us time to discuss this specifically. It will be related to objective's definition; (min, max), (deterministic, noisy), etc.